crates/core/tedge_api/src/entity_store.rs

```diff
@@ -168,13 +258,13 @@ impl EntityStore {
             affected_entities.push(parent.clone());
         }
 
-        let entity_id = message
-            .entity_id
-            .unwrap_or_else(|| self.derive_entity_id(&message.topic_id));
+        let external_id = message.entity_id.unwrap_or_else(|| {
```

Comment:
There was a suggestion from @reubenmiller to apply the same transformation rules(replace `/` with `:`) to the user provided `@id` value as well, rather than just using it as-is, which could mess up the external ID in SmartREST topics. I was initially thinking of just treating it as bad input if it contains "special forbidden chars" like `/, +, #`. But, this proposed scheme might be a bit more flexible, with the caveat of us overrriding even user overridden value. So,  @didier-wenzek, thoughts?